server/src/main/java/org/elasticsearch/search/DefaultSearchContext.java

```diff
@@ -468,8 +467,8 @@ public IndexShard indexShard() {
     }
 
     @Override
-    public MapperService mapperService() {
-        return indexService.mapperService();
+    public IndexSettings indexSettings() {
```

Comment:
I was also going to add this but then I figured that we have a getter for the shard context which already exposes the settings, so I went for calling that one directly.

Reply:
Makes sense. I'm happy to drop it.